### PR TITLE
Finished reviewing module 4 and the appendix

### PR DESCRIPTION
**Pull request by waelkdouh**
Code review for module 4 and the appendix module.

### CODE REVIEW

File: docs/aca/04-aca-dapr-stateapi/index.md [hunk 1]
```diff
@@ -94,9 +109,12 @@ Similar to what we have done in the Frontend Web App, we need to use Dapr Client
 ```
 
 ##### 2. Create a new concrete implementation to manage tasks persistence
-As you recall from the previous module, we were storing the tasks in memory, now we need to store them in Redis and later on Azure Cosmos DB. To do so add a new file named `TasksStoreManager.cs` under the folder named `Services` and this file will implement the interface `ITasksManager`. Copy & Paste the code below:
+As you recall from the previous module, we were storing the tasks in memory. Now we need to store them in Redis and later on Azure Cosmos DB. The key thing to keep in mind here is that switching from redis to Azure Cosmos DB won't required changing the code below which is a huge advantage of using Dapr. Add a new file named `TasksStoreManager.cs` under the folder named `Services`. This file will implement the interface `ITasksManager`. Copy & Paste the code below:
```
Comment (tjoudeh) [inline docs/aca/04-aca-dapr-stateapi/index.md:112]: won't require

File: docs/aca/04-aca-dapr-stateapi/index.md [hunk 2]
```diff
@@ -194,7 +212,7 @@ The query API will not work against the local Redis store as you need to install
 
 ##### 3. Register the TasksStoreManager new service and DaprClient
 
-Now we need to register the new service named `TasksStoreManager` and `RedisClient` when the Backend API app starts up, to do so open the file `Program.cs` and register both as the below. Do not forget to comment out the registration of the `FakeTasksManager` service as we don’t want to store tasks in memory anymore.
+Now we need to register the new service named `TasksStoreManager` and `RedisClient` when the Backend API app starts up. To do so open the file `Program.cs` and register both as shown below. Do not forget to comment out the registration of the `FakeTasksManager` service as we don’t want to store tasks in memory anymore.
```
Comment (tjoudeh) [inline docs/aca/04-aca-dapr-stateapi/index.md:215]: `DaprClient`

File: docs/aca/04-aca-dapr-stateapi/index.md [hunk 3]
```diff
@@ -291,17 +310,39 @@ scopes:
...
+- Replace the url value with the URI value of your cosmos database account. You can get that from the Azure portal by navigating to the cosmos database account overview page and get the uri value from there. Basically the uri should have the following structure. https://COSMOS_DB_ACCOUNT.documents.azure.com:443/.
+- By default, all dapr-enabled container apps within the same environment will load the full set of deployed components. By adding `scopes` to a component, you tell the Dapr sidecars for each respective container app which components to load at runtime. Using scopes is recommended for production workloads. In our case, we have set the scopes to `tasksmanager-backend-api` as this will be the application that needs access to Azure Cosmos DB State Store. More about scopes can be found on this [link](https://learn.microsoft.com/en-us/azure/container-apps/dapr-overview?tabs=bicep1%2Cyaml#component-scopes).
```
Comment (tjoudeh) [inline docs/aca/04-aca-dapr-stateapi/index.md:314]: Maybe rephrase to highlight that value `tasksmanager-backend-api` is the dapr app id not the container app name, we can use the following: 

we have set the scopes to `tasksmanager-backend-api` which represents the `dapr-app-id` which is associated to the container app that needs access to Azure Cosmos DB State Store.

File: docs/aca/04-aca-dapr-stateapi/index.md [hunk 4]
```diff
@@ -291,17 +310,39 @@ scopes:
...
 
 {: .note }
 Dapr component scopes correspond to the Dapr application ID of a container app, not the container app name.
 
-Now you should be ready to launch both applications and start doing CRUD operations from the Frontend Web App including querying the store, all your data will be stored in Cosmos DB Database you just provisioned, it should look like the below:
+Now you should be ready to launch both applications and start doing CRUD operations from the Frontend Web App including querying the store. All your data will be stored in Cosmos DB Database you just provisioned. If you have been running the different microservices using the [debug and launch Dapr applications in VSCode](../../aca/20-appendix/01-run-debug-dapr-app-vscode.md) then remember to uncomment the following line inside tasks.json file. This will dapr to load the local projects components instead of the global components folder.
```
Comment (tjoudeh) [inline docs/aca/04-aca-dapr-stateapi/index.md:319]: This will _instruct_ dapr to load the local projects components (./components)

File: docs/aca/04-aca-dapr-stateapi/index.md [hunk 5]
```diff
@@ -291,17 +310,39 @@ scopes:
...
+```json
+"componentsPath": "./components"
+```
+
+If you have been using the dapr cli commands instead of the aforementioned debugging then you will need to execute the backend api with the components-path property as follows. Remember to replace the place holders with your own values:
```
Comment (tjoudeh) [inline docs/aca/04-aca-dapr-stateapi/index.md:324]: you will need to execute the backend api with the _resources-path_ property?

**Review by tjoudeh — COMMENTED** [review]
minor comments, please check and complete PR

**Review by tjoudeh — APPROVED** [review]

File: docs/aca/04-aca-dapr-stateapi/index.md (hunk 2 above)
Comment (waelkdouh) [reply docs/aca/04-aca-dapr-stateapi/index.md:215]: Not sure what you mean by DaprClient. Can you please further clarify this?
